modules/repository/cache.go

```diff
@@ -84,14 +41,7 @@ func CacheRef(repo *models.Repository, gitRepo *git.Repository, fullRefName stri
 		return nil
 	}
 
-	commitNodeIndex, _ := gitRepo.CommitNodeIndex()
-
-	c, err := commitNodeIndex.Get(commit.ID)
-	if err != nil {
-		return err
-	}
-
-	ca := cache.NewLastCommitCache(repo.FullName(), gitRepo, int64(setting.CacheService.LastCommit.TTL.Seconds()))
+	commitCache := git.NewLastCommitCache(repo.FullName(), gitRepo, int64(setting.CacheService.LastCommit.TTL.Seconds()), cache.GetCache())
 
-	return recusiveCache(gitRepo, c, &commit.Tree, "", ca, 1)
+	return commitCache.CacheCommit(gitRepo, commit)
```

Comment:
The first parameter could be ignored since commitCache should have chance to store it.